src/vm/docs/mod.rs

```diff
@@ -619,6 +619,36 @@ is supplied the hash is computed over the little-endian representation of the in
     example: "(keccak256 0) ;; Returns 0xf490de2920c8a35fabeb13208852aa28c76f9be9b03a4dd2b3c075f7a26923b4"
 };
 
+const SECP256K1RECOVER_API: SpecialAPI = SpecialAPI {
+    input_type: "(buff 32), (buff 65)",
+    output_type: "(response (buff 33) uint)",
+    signature: "(secp256k1-recover? message-hash signature)",
+    description: "The `secp256k1-recover?` function recovers the public key used to sign the message  which sha256 is `message-hash`
+    with the provided `signature`.
+    If the signature does not match, it will return the error code `(err u1).`.
+    If the signature is invalid, it will return the error code `(err u2).`.
+    The signature includes 64 bytes plus an additional recovery id (00..03) for a total of 65 bytes.",
+    example: "(secp256k1-recover? 0xde5b9eb9e7c5592930eb2e30a01369c36586d872082ed8181ee83d2a0ec20f04
+ 0x8738487ebe69b93d8e51583be8eee50bb4213fc49c767d329632730cc193b873554428fc936ca3569afc15f1c9365f6591d6251a89fee9c9ac661116824d3a1301)
+ ;; Returns (ok 0x03adb8de4bfb65db2cfd6120d55c6526ae9c52e675db7e47308636534ba7786110)"
+};
+
+const SECP256K1VERIFY_API: SpecialAPI = SpecialAPI {
+    input_type: "(buff 32), (buff 65), (buff 33)",
```

Comment:
I missed this the first time. In this case, we're verifying, meaning we should be dealing with a "standard" signature `buff 64` instead of a `buff 65`, no?

Reply:
this is intentional so the same signature (which includes the recovery id as the last byte) can be used for both `secp256k1-verify` and `secp256k1-recover`.  If the hash was signed with `sign` and not `sign_recoverable`,  `0x00` can be added.

We could make it so that both 64 bytes and 65 bytes could be passed to `secp256k1-verify` but the 64 bytes version would fail if you pass it to `secp356k1-recover`.

I think just having to use 65 bytes keeps things simpler.

Reply:
In my experience, either you sign for verify, or you sign(_recoverable) for recover, but I'm having a hard time envisioning a use case in a smart contract, where you would use a verify and recover in the same sequence, and with the same signature.
In my mind, it's easier and simpler for a developer to just throw 1) the output of `sign` to `verify`, and 2) the output of `sign_recoverable` to `recover`, vs adding one useless extra byte in case 1.

Reply:
I changed the function so it can take either a 64 or 65 bytes signatures, which should worth whether the signature includes the recoveryId or not (see 6db1099a446a963d45655f2ab59028fb1b32e19f)